### PR TITLE
feat: response interception

### PR DESCRIPTION
**Pull request by yury-s**
Initial API and implementation for Chromium and WebKit.

#1774

### CODE REVIEW

File: src/server/chromium/crNetworkManager.ts [hunk 1]
```diff
@@ -374,14 +387,63 @@ export class CRNetworkManager {
...
+  private _responseCode: number;
+  private _responseErrorReason: Protocol.Network.ErrorReason | undefined;
+  private _responseHeaders: Protocol.Fetch.HeaderEntry[];
+
+  constructor(request: InterceptableRequest, event: Protocol.Fetch.requestPausedPayload) {
+    this._request = request;
+    this._responseCode = event.responseStatusCode!;
+    this._responseErrorReason = event.responseErrorReason;
+    this._responseHeaders = event.responseHeaders!;
+}
```
Comment (aslushnikov) [inline src/server/chromium/crNetworkManager.ts:401]: nit: spaces

File: src/server/chromium/crNetworkManager.ts [hunk 2]
```diff
@@ -374,14 +387,63 @@ export class CRNetworkManager {
...
+        body = Buffer.from(overrides.body).toString('base64');
+    } else {
+      body = (await this.body()).toString('base64');
+    }
+    await this._request._client._sendMayFail('Fetch.fulfillRequest', {
+      requestId: this._request._interceptionId!,
+      responseCode: overrides.status || this._responseCode,
+      responsePhrase: overrides.statusText || network.STATUS_TEXTS[String(this._responseCode)],
+      responseHeaders: overrides.headers || this._responseHeaders,
+      body,
```
Comment (aslushnikov) [inline src/server/chromium/crNetworkManager.ts:428]: nit: a little scary to always pump body over protocol

**Review by aslushnikov — APPROVED** [review]
!!!

File: src/client/connection.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ import { ChannelOwner } from './channelOwner';
 import { ElementHandle } from './elementHandle';
 import { Frame } from './frame';
 import { JSHandle } from './jsHandle';
-import { Request, Response, Route, WebSocket } from './network';
+import { InterceptedResponse, Request, Response, Route, WebSocket } from './network';
```
Comment (aslushnikov) [inline src/client/connection.ts:24]: stray change?

**Review by aslushnikov — APPROVED** [review]

File: src/client/connection.ts (hunk 3 above)
Comment (yury-s) [reply src/client/connection.ts:24]: yep

**Pull request merged**